tests/nodes/connections.test.ts

```diff
@@ -0,0 +1,129 @@
+import type { NodeId, NodeIdEncoded } from 'polykey/dist/ids/types';
+import path from 'path';
+import fs from 'fs';
+import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
+import PolykeyAgent from 'polykey/dist/PolykeyAgent';
+import * as nodesUtils from 'polykey/dist/nodes/utils';
+import * as keysUtils from 'polykey/dist/keys/utils';
+import * as testUtils from '../utils';
+
+describe('connections', () => {
+  const logger = new Logger('connections test', LogLevel.WARN, [
+    new StreamHandler(),
+  ]);
+  const password = 'helloworld';
+  let dataDir: string;
+  let nodePath: string;
+  let pkAgent: PolykeyAgent;
+  let remoteNode: PolykeyAgent;
+  let localId: NodeId;
+  let remoteId: NodeId;
+  let remoteIdEncoded: NodeIdEncoded;
+  beforeEach(async () => {
+    dataDir = await fs.promises.mkdtemp(
+      path.join(globalThis.tmpDir, 'polykey-test-'),
+    );
+    nodePath = path.join(dataDir, 'keynode');
+    pkAgent = await PolykeyAgent.createPolykeyAgent({
+      password,
+      options: {
+        seedNodes: {}, // Explicitly no seed nodes on startup
+        nodePath,
+        agentServiceHost: '127.0.0.1',
+        clientServiceHost: '127.0.0.1',
+        agentServicePort: 55555,
+        clientServicePort: 55554,
```

Comment:
Can these ports be `0` and just dynamically refer to them? I avoid hardcoding ports even in tests.

Reply:
this has been changed, no need to dynamically refer to them, as the nodepaths are used in the tests already